node.gyp

```diff
@@ -498,6 +498,7 @@
         'src/node_contextify.cc',
         'src/node_credentials.cc',
         'src/node_dir.cc',
+        'src/node_encoding.cc',
```

Comment:
Nit: we already have so many files dealing with encodings, does this really need to be a new one (instead of, e.g., staying in `node_buffer` or maybe `string_bytes`)? And if so, is this _the_ encoding file?

Reply:
I'm still not sure why this is a new file when it only covers a tiny part of all encoding-related routines.